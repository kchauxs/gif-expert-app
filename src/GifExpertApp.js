import React, { Fragment, useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'
import { CleanButton } from './components/CleanButton'

export const GifExpertApp = ({defaultCategories = []}) => {

    const [categories, setCategoties] = useState(defaultCategories)

    return (
        <Fragment>
            <h2>GifExpertApp {":)"}</h2>
            <AddCategory setCategoties={setCategoties} />
            <CleanButton setCategoties={setCategoties} />
            <hr />
            <ol>
                {categories.map(cat => <GifGrid category={cat} key={cat} />)}
            </ol>
        </Fragment>
    )
}
