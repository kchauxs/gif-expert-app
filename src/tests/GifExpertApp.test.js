import React from 'react'
import "@testing-library/jest-dom"
import { shallow } from "enzyme"
import {GifExpertApp} from "../GifExpertApp"


describe('<GifExpertApp/>', () => {


    test('should: mostrar correctamente ', () => {
        const wrapper = shallow(<GifExpertApp/>)
        expect(wrapper).toMatchSnapshot()
    })


    test('should ', () => {
        const categories = ["boo", "rick"]
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find("GifGrid").length).toBe(categories.length)

    })
    

})

