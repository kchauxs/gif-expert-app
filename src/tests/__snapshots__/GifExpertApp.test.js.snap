// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp/> should  1`] = `
<Fragment>
  <h2>
    GifExpertApp 
    :)
  </h2>
  <AddCategory
    setCategoties={[Function]}
  />
  <CleanButton
    setCategoties={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="boo"
      key="boo"
    />
    <GifGrid
      category="rick"
      key="rick"
    />
  </ol>
</Fragment>
`;

exports[`<GifExpertApp/> should: mostrar correctamente  1`] = `
<Fragment>
  <h2>
    GifExpertApp 
    :)
  </h2>
  <AddCategory
    setCategoties={[Function]}
  />
  <CleanButton
    setCategoties={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
